### PR TITLE
[ISSUE-#4066] Let the service's order followed the order which it be added.

### PR DESCRIPTION
**Pull request by horizonzy**
## What is the purpose of the change

Fix-#4066

### CODE REVIEW

**Comment by KomachiSion** [comment]
Will `Collections.synchronizedMap` reduce performance than the `ConcurrentHashMap`?

**Comment by horizonzy** [comment]
> Will `Collections.synchronizedMap` reduce performance than the `ConcurrentHashMap`?

It will, it's exclusive lock. I will try another way to realize it, just pending it

**Comment by horizonzy** [comment]
@chuntaojun @KomachiSion

File: istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosMcpService.java [hunk 1]
```diff
@@ -94,14 +95,14 @@ public void run() {
             Set<String> namespaces = serviceManager.getAllNamespaces();
             
             for (String namespace : namespaces) {
-                
-                Map<String, Service> services = serviceManager.chooseServiceMap(namespace);
-                
-                if (services.isEmpty()) {
+    
+                Collection<Service> values = serviceManager.chooseServiceMap(namespace).values();
```
Comment (chuntaojun) [inline istio/src/main/java/com/alibaba/nacos/istio/mcp/NacosMcpService.java:99]: Here directly use forEach to traverse, don't use values()

File: naming/src/main/java/com/alibaba/nacos/naming/core/ServiceManager.java [hunk 2]
```diff
@@ -1135,6 +1140,49 @@ public void run() {
         }
     }
     
+    private static class ServiceNameHelper implements Comparable<ServiceNameHelper> {
```
Comment (chuntaojun) [inline naming/src/main/java/com/alibaba/nacos/naming/core/ServiceManager.java:1143]: The meaning of this class? I feel that with this design, I am very puzzled. Is there a Map inside ServiceNameHelper?

**Comment by horizonzy** [comment]
This pr can't ensure the global order in cluster.

**Pull request closed without merging**